piecedirectory/test_util.go

```diff
@@ -0,0 +1,118 @@
+package piecedirectory
```

Comment:
Nit but maybe name this file `util_test.go` so that it is not compiled in the default binary if these are only used for tests.

Reply:
Unfortunately I don't think we can use the methods from a different package if the filename ends with `_test.go`